### PR TITLE
Add JDK11 support and enable in CI

### LINKED ISSUE #31230: Upgrade to Gradle 4.8 and enable JDK 11 in CI
With rampdown phase one of JDK 11 approaching, we want to enable JDK 11 in CI. However, Gradle 4.7 is not compatible with JDK 11 (gradle/gradle#4860). The upgrade to Gradle 4.8 is non-trivial due to Gradle reporting dependency conflicts. To enable JDK 11 in CI we need:
 - [ ] upgrade the Gradle wrapper to 4.8
 - [ ] enable JDK 11 to be used as the compiler (while still allowing JDK 10)
 - [ ] get the build passing on JDK 11
 - [ ] add JDK 11 to the various properties files in `.ci`
 - [ ] work with our internal infra team to have the EA build of the JDK upgraded to JDK 11-latest in CI

These changes should be done for 6.x and master.

### PR DESCRIPTION
**Pull request by alpar-t**
- Update CI to include JDK 11. 
- mute failing tests with issue ID mentioned
- upgrade bouncycastle to avoid invalid manifest errors on JDK 11
- improve  how BWC builds are logged. This one is not directly related to JDK 11, but helped debug failures while working on it. Each line of the BWC build will be prefixed and indented, to make it easier to distinguish from CI logs that it's the BWC build and not the main Gradle build.

Closes #31230,  #31500

### CODE REVIEW

File: .ci/java-versions.properties [hunk 1]
```diff
@@ -4,5 +4,5 @@
 # build and test Elasticsearch for this branch. Valid Java versions
 # are 'java' or 'openjdk' followed by the major release number.
 
-ES_BUILD_JAVA=java10
-ES_RUNTIME_JAVA=java8
+ES_BUILD_JAVA=java11
```
Comment (atorok) [inline .ci/java-versions.properties]: @jasontedor I'm actually not sure if this is ok, or we should have it only for matrix and do this lather on.

Comment (jasontedor) [reply .ci/java-versions.properties]: Indeed, we need to hold off on this specific change and leave it to the matrix only for now until we reach conclusion on the minimum runtime JDK for 7.0.0.

**Comment by atorok** [comment]
I applied the v6.4.0 label, to back-port this PR -CI changes and muted tests. 
Will do it with a separate PR.

**Review by nik9000 — COMMENTED** [review]
I hate having all of the awaitsfixes but that is life I guess. Beyond those this looks good to me.

I suppose we assign those failing tests to some owner?

**Comment by jasontedor** [comment]
Can we not awaits fix those and instead make them assumeFalse(JDK 11)?

**Review by jasontedor — COMMENTED** [review]
I left one comment. I am not marking it as request changes because I will not be around next week to approve the change. I would lean on someone else to pick up this thread.

File: buildSrc/src/main/resources/forbidden/jdk-signatures.txt [hunk 2]
```diff
@@ -88,7 +88,6 @@ java.lang.Thread#getAllStackTraces()
 
 @defaultMessage Stopping threads explicitly leads to inconsistent states. Use interrupt() instead.
 java.lang.Thread#stop()
-java.lang.Thread#stop(java.lang.Throwable)
```
Comment (rjernst) [inline buildSrc/src/main/resources/forbidden/jdk-signatures.txt]: Why is this removed from forbidden apis? The forbidden signatures should against our minimum runtime, which is still java 8.

File: distribution/bwc/build.gradle [hunk 3]
```diff
@@ -196,3 +204,25 @@ subprojects {
     }
   }
 }
+
+class IdentingOutputStream extends OutputStream {
```
Comment (rjernst) [inline distribution/bwc/build.gradle]: typo: Identing -> Indenting

File: distribution/bwc/build.gradle [hunk 4]
```diff
@@ -196,3 +204,25 @@ subprojects {
...
+
+  @Override
+  public void write(int b) {
+    write([b] as int[], 0, 1)
+  }
+
+  public void write(int[] bytes, int offset, int length) {
+    delegate.write(
+            new String(bytes, offset, length)
+                    .replace("\n", "\n  [bwc] ")
```
Comment (rjernst) [inline distribution/bwc/build.gradle]: I think this is going to be pretty inefficient. Instead, you can loop over the bytes looking for newline. When you find a newline, emit the array up to that point. Then emit a fixed byte array you have with the newline and prefix. Then continue the loop. You won't need to create any temporary objects.

**Review by rjernst — COMMENTED** [review]
This looks good. I left a few comments. Additionally, I think any issues that are linked here as TODOs need to be made blockers so that they are at least investigated and their impact assessed before 6.4.0.

File: buildSrc/src/main/resources/forbidden/jdk-signatures.txt (hunk 2 above)
Comment (atorok) [reply buildSrc/src/main/resources/forbidden/jdk-signatures.txt]: The method was removed in jdk 11 and causing the test to fail. 
Do we have per JDK version forbidden APIs ?

Comment (rjernst) [reply buildSrc/src/main/resources/forbidden/jdk-signatures.txt]: We do not. I think this is related to running forbidden apis in a separate jvm. We should always be running it with the minimum supported version.

Comment (rjernst) [reply buildSrc/src/main/resources/forbidden/jdk-signatures.txt]: Can you please comment this out and reference the issue to run forbidden apis in a forked jvm.

Comment (uschindler) [reply buildSrc/src/main/resources/forbidden/jdk-signatures.txt]: If this issue was the only reason to no longer use the gradle plugin and fork instead, I can just say: The above forbidden-api signature can be removed anyways from elasticsearch (together with the previous one). Forbiddenapis has those signatures already in its "jdk-deprecated" ones, as it's a known problem - so there is no need to have it in Elasticsearch's custom signatures. "jdk-deprecated" automatically handles deprecations that were removed.

IMHO, #31715 was not a good idea, sorry!

Comment (uschindler) [reply buildSrc/src/main/resources/forbidden/jdk-signatures.txt]: https://github.com/policeman-tools/forbidden-apis/blob/master/src/main/resources/de/thetaphi/forbiddenapis/signatures/jdk-deprecated-1.8.txt#L175-L180

and

https://github.com/policeman-tools/forbidden-apis/blob/master/src/main/resources/de/thetaphi/forbiddenapis/signatures/jdk-deprecated-1.8.txt#L7
